### PR TITLE
[webapp]:Add endpoint for AppService CLI Github token endpoint

### PR DESCRIPTION
**Pull request by calvinsID**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

- Add endpoint for AppService CLI Github token endpoint
- Update WebSiteManagementClient default tag to 2020-06

ARM Manifest change PR: https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/3700170

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/web/resource-manager/readme.md
Comment (calvinsID) [inline specification/web/resource-manager/readme.md:37]: For task: Making 2020-06-01 the default api-version (https://msazure.visualstudio.com/Antares/_sprints/taskboard/Domains%20and%20Certificates/Antares/UX/s150?workitem=8469099). @naveedaz are there any other changes needed for this?

File: specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/ResourceProvider.json
Comment (markcowl) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/ResourceProvider.json:18]: It is not allowed to add a new api into an existing api-version.  Please see the breaking change document (and linked API guidelines) from the checklist in PR comments.

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by markcowl** [comment]
@naveedaz  Please fill out the contribution checklist and read the associated documentation.  It is no longer allowed to add new APIs into an existing API Version

**Comment by panchagnula** [comment]
> @naveedaz Please fill out the contribution checklist and read the associated documentation. It is no longer allowed to add new APIs into an existing API Version

@markcowl could you point us to the documentation that you are referring to here? We found this https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/82/Manual-Process-Adding-new-resource-type-or-APIs-to-an-existing-API-version - and so were not aware that we are required to have a new api-version for this change. Thanks!

**Comment by markcowl** [comment]
@naveedaz: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#api-changes-that-require-a-version-change

**Comment by markcowl** [comment]
@calvinsID Please fix the linter errors (and warnings) indicated in the LintDiff failure above.  Also, this will need an ARM review, as indicated.

**Comment by calvinsID** [comment]
@markcowl Regarding the R4015 linter errors, they are in the previous 2020-06 version and unrelated to any changes. Can I get approval to bypass them?

**Comment by panchagnula** [comment]
@akning-ms can you please review this change please?

**Review by panchagnula — APPROVED** [review]
LGTM. @naveedaz could you review as well.

**Comment by panchagnula** [comment]
@akning-ms can we get a review/signoff on this please?

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/CommonDefinitions.json
Comment (naveedaz) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/CommonDefinitions.json:1]: Why is this file needed?

Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/CommonDefinitions.json:1]: I added it because the ResourceProvider.json file references "./CommonDefinitions.json#/definitions" in many areas

Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/CommonDefinitions.json:1]: @naveedaz although I did realise I forgot to bump the version of CommonDefinitions in the readme... fixed. Please lemme know if I should do otherwise!

Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/CommonDefinitions.json:1]: also @naveedaz do you know what the cause of the Linter Error is after bumping the CommonDefinitions in the readme? I can't seem to figure it out from the error msg

**Review by naveedaz — APPROVED** [review]

**Comment by panchagnula** [comment]
> @markcowl Regarding the R4015 linter errors, they are in the previous 2020-06 version and unrelated to any changes. Can I get approval to bypass them?

@markcowl can we get a review on the updates please. Thanks!

**Comment by panchagnula** [comment]
Ping @akning-m

**Comment by akning-ms** [comment]
> Ping @akning-ms

Hi @panchagnula , This PR also need ARM Sign off.  and this PR is adding a new stable version 2020-09-01. but only includes ResourceProvider.json, our best practice is every version include all your API which like version 2020-06-01. can we know why? 
And can you give us for info for this PR for us better review.
1. what is the purpose of the PR?
2. when your service is ready on production
3. when your swagger target to merge?
4. Is there any SDK or CLI that require refresh after Swagger published?

**Comment by calvinsID** [comment]
Hi @akning-ms, who would be able to sign off on behalf of ARM? We only bumped ResourceProvider to 2020-09 because this swagger update only includes update to a single endpoint in ResourceProvider. To answer your questions:
1. The PR is to include the endpoint POST `/providers/Microsoft.Web/generateGithubAccessTokenForAppserviceCLI`, which is needed by Azure CLI. It is used as part of CLI's OAuth flow with github - it takes in a code, and returns to the user a github personal access token
2. The endpoint was released as part of ANT89
3. We want to merge the swagger as soon as possible
4. We want to generate python SDK so that CLI can use this

Please let me know if you have any more questions!

**Comment by akning-ms** [comment]
> Hi @akning-ms, who would be able to sign off on behalf of ARM? We only bumped ResourceProvider to 2020-09 because this swagger update only includes update to a single endpoint in ResourceProvider. To answer your questions:
> 
> 1. The PR is to include the endpoint POST `/providers/Microsoft.Web/generateGithubAccessTokenForAppserviceCLI`, which is needed by Azure CLI. It is used as part of CLI's OAuth flow with github - it takes in a code, and returns to the user a github personal access token
> 2. The endpoint was released as part of ANT89
> 3. We want to merge the swagger as soon as possible
> 4. We want to generate python SDK so that CLI can use this
> 
> Please let me know if you have any more questions!

If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:47]: Can you please add CommonDefinitions.json for 2020-09-01?

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:668]: Since access token should be treated as a secret, please add `"x-ms-secret": true`.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:682]: AppserviceGithubToken is defined for successful operation (Http code 200). Why does it need an error message? Error should be returned as a common ARM error object with an error code.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:696]: Is this a secret? Then please set `x-ms-secret` to true.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:18]: Do you also have ARM manifest changes to support this action? Can you please provide a PR link?

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/examples/VerifyHostingEnvironmentVnet.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/examples/VerifyHostingEnvironmentVnet.json:16]: Please provide examples for all APIs exposed by 2020-09-01 including one for the new /providers/Microsoft.Web/generateGithubAccessTokenForAppserviceCLI API.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:18]: ARM Manifest change: 
https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/3700170?_a=files&path=%2Fsrc%2FHosting%2FAdministrationService%2FMicrosoft.Web.Hosting.Administration.GeoScale.Csm%2FCSMManifests%2FMicrosoft.Web.Manifest-prod.json

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:47]: Is it ok to keep using the 2020-06-01 CommonDefinitions? Since there are no changes to it, and we don't want to increment anything we don't need to

Comment (j5lim) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:47]: I'm not sure if there is official guideline for this, but I think all models representing an api-version should be maintained under the api-version. It's easier to maintain (i.e. don't need to cross-check multiple api-versions when updating the model).

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:18]: Thank you. generateGithubAccessTokenForAppserviceCLI is modeled as a tracked resource in the manifest. Tracked resources must implement GET, PUT, PATCH, DELETE & LIST. 
https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc006-tracked-resource-type-supports-get-put-patch-delete--list-at-all-scopes-as-applicable

I think it should be a proxy resource.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:661]: As of API version 2020-05-01, resources must begin adding the common systemData object property to the top-level property set on read responses (responses for PUT, GET, and PATCH).
RPC: https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc027-systemdata-support

You can reference [common type](https://github.com/Azure/azure-rest-api-specs/blob/20283a103ef5216f805fccc7ace6b6fc26c0cfbd/specification/common-types/resource-management/v2/types.json#L445) or have a copy of it.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:18]: good catch, thanks. Changed to proxy

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/examples/VerifyHostingEnvironmentVnet.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/examples/VerifyHostingEnvironmentVnet.json:16]: Added

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json]: Added

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:687]: It's probably better to have named it status message. ARM error object with error code would be returned for errors

File: specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/ResourceProvider.json]: Updated to new version, thanks

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:701]: Added

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json]: I'm getting validation error saying secret cannot be sent as a response when I add this, but getting this access token is the purpose of this endpoint
image

Comment (j5lim) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:673]: POST can return a secret. Looks like we need to update validation. It's okay not to have x-ms-secret annotation for now.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:18]: Updated manifest since we're bumping all API's now

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:666]: Added

File: specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json:691]: Sorry I should have clarified. Can you please add systemData to both "Resource" and "ProxyOnlyResource" definitions? That way all tracked & proxy resources will have it.

Comment (calvinsID) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-09-01/ResourceProvider.json]: Updated

**Review by j5lim — APPROVED** [review]

**Comment by akning-ms** [comment]
Hi @calvinsID,  some properties(vnetRouteAllEnabled..) removed in new version. this could be a breaking change.
refer to: https://github.com/Azure/azure-rest-api-specs/blob/33b5733cc82028c9ce6e2c9ec9f28e7f763098f7/specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/CommonDefinitions.json#L2152:L2160

**Comment by calvinsID** [comment]
> Hi @calvinsID, some properties(vnetRouteAllEnabled..) removed in new version. this could be a breaking change.
> refer to: https://github.com/Azure/azure-rest-api-specs/blob/33b5733cc82028c9ce6e2c9ec9f28e7f763098f7/specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/CommonDefinitions.json#L2152:L2160



> Hi @calvinsID, some properties(vnetRouteAllEnabled..) removed in new version. this could be a breaking change.
> refer to: https://github.com/Azure/azure-rest-api-specs/blob/33b5733cc82028c9ce6e2c9ec9f28e7f763098f7/specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/CommonDefinitions.json#L2152:L2160

Good catch, thanks! Seems like this was added in after I copied over the old (2020-06) version. Will add back into new (2020-09) version

**Pull request merged**
